### PR TITLE
[Torch Dialect] support aten.isneginf, aten.isposinf, aten.nan_to_num

### PR DESCRIPTION
**Pull request by linuxlonelyeagle**

### CODE REVIEW

File: lib/Dialect/Torch/Transforms/DecomposeComplexOps.cpp [hunk 1]
```diff
@@ -2471,6 +2505,49 @@ class DecomposeAtenWhereScalarSelfOp
...
 
+namespace {
+class DecomposeAtenNanToNumOp : public OpRewritePattern<AtenNanToNumOp> {
+public:
+  using OpRewritePattern::OpRewritePattern;
+  LogicalResult matchAndRewrite(AtenNanToNumOp op,
+                                PatternRewriter &rewriter) const override {
+    Location loc = op.getLoc();
+    mlir::FloatType f64Type = rewriter.getF64Type();
+    Value lan = op.getNan();
```
Comment (qingyunqu) [inline lib/Dialect/Torch/Transforms/DecomposeComplexOps.cpp:2516]: nit: `Value lan` => `Value nan`

**Review by qingyunqu — CHANGES_REQUESTED** [review]

File: projects/pt1/python/torch_mlir_e2e_test/test_suite/elementwise.py [hunk 2]
```diff
@@ -3463,6 +3490,58 @@ def ElementwiseAtenIsinfOpModule_basic(module, tu: TestUtils):
...
+    ])
+    def forward(self, x):
+        return torch.ops.aten.isneginf(x)
+
+@register_test_case(module_factory=lambda: ElementwiseAtenIsneginfOpModule())
+def ElementwiseAtenIsneginfOpModule_basic(module, tu:TestUtils):
+    test_input = torch.tensor(
+        [
+            [1, float('-inf'), 2, float('-inf'), float('nan')],
+            [1, float('-inf'), float('-inf'), float('nan'), 3],
```
Comment (qingyunqu) [inline projects/pt1/python/torch_mlir_e2e_test/test_suite/elementwise.py:3510]: change some `-inf` to `inf` to make this test more robust? Same as test of `aten.isposinf`.

**Review by qingyunqu — APPROVED** [review]
LGTM

**Comment by qingyunqu** [comment]
@ramiro050 Could you please review this?

**Review by Vremold — APPROVED** [review]
LGTM. Thanks for the effort.

**Pull request merged**